### PR TITLE
feat(dump): Provide the same cli options as the snapshots

### LINKED ISSUE #2087: New dump behavior
Following this spec comment: https://github.com/meilisearch/specifications/pull/105#issuecomment-1014655194

New behavior regarding the dump:

- [ ] When importing a dump using `--import-dump` if a database already exists, an error message will be thrown on the CLI. The message is `Error: database already exists at ":pathToDataMs/data.ms", try to delete it or rename it.`.

- [ ] Similar to the behavior of the snapshot feature. The flags --ignore-dump-if-db-exists and --ignore-missing-dump are added.

Check the whole spec for more details: https://github.com/meilisearch/specifications/pull/105/files

Impacts @meilisearch/docs-team but also @meilisearch/integration-team for the `migration-script`.

### PR DESCRIPTION
**Pull request by irevoire**
Add two cli options for the dump:
- `--ignore-missing-dump`
- `--ignore-dump-if-db-exists`

Fix #2087

### CODE REVIEW

File: meilisearch-http/src/option.rs [hunk 1]
```diff
@@ -112,14 +112,22 @@ pub struct Opt {
...
-
     /// Import a dump from the specified path, must be a `.dump` file.
     #[clap(long, conflicts_with = "import-snapshot")]
     pub import_dump: Option<PathBuf>,
 
+    /// The engine will ignore a missing dump and not return an error in such case.
+    #[clap(long, requires = "import-dump")]
+    pub ignore_missing_dump: bool,
+
+    /// The engine will skip dump importation and not return an error in such case.
```
Comment (MarinPostma) [inline meilisearch-http/src/option.rs:123]:
```suggestion
    /// Ignore the dump if a database already exists, and load that database instead.
```

File: meilisearch-http/src/option.rs [hunk 2]
```diff
@@ -112,14 +112,22 @@ pub struct Opt {
...
 
-    /// Folder where dumps are created when the dump route is called.
-    #[clap(long, env = "MEILI_DUMPS_DIR", default_value = "dumps/")]
-    pub dumps_dir: PathBuf,
-
     /// Import a dump from the specified path, must be a `.dump` file.
     #[clap(long, conflicts_with = "import-snapshot")]
     pub import_dump: Option<PathBuf>,
 
+    /// The engine will ignore a missing dump and not return an error in such case.
```
Comment (MarinPostma) [inline meilisearch-http/src/option.rs:119]:
```suggestion
    /// If the dump doesn't exists, load or create the database specified by `db-path` instead.
```

File: meilisearch-lib/src/lib.rs [hunk 3]
```diff
@@ -33,3 +35,19 @@ impl EnvSizer for heed::Env {
             .fold(0, |acc, m| acc + m.len())
     }
 }
+
+/// Check if a db is empty. It does not provide any information on the
+/// validity of the data in it.
+/// We consider a database as non empty when it's a non empty directory.
+pub fn empty_db(db_path: impl AsRef<Path>) -> bool {
```
Comment (MarinPostma) [inline meilisearch-lib/src/lib.rs:42]:
```suggestion
pub fn is_empty_db(db_path: impl AsRef<Path>) -> bool {
```

File: meilisearch-lib/src/index_controller/dump_actor/mod.rs [hunk 4]
```diff
@@ -160,10 +161,31 @@ impl DumpInfo {
...
+    ignore_missing_dump: bool,
     index_db_size: usize,
     update_db_size: usize,
     indexer_opts: &IndexerOpts,
 ) -> anyhow::Result<()> {
+    let empty_db = crate::empty_db(&dst_path);
+    let src_path_exists = src_path.as_ref().exists();
+
+    if empty_db && src_path_exists {
+        // everything is fine, there is no early return
```
Comment (MarinPostma) [inline meilisearch-lib/src/index_controller/dump_actor/mod.rs:174]: can you instead move the relevant code in this branch?

**Review by MarinPostma — CHANGES_REQUESTED** [review]

File: meilisearch-lib/src/index_controller/dump_actor/mod.rs (hunk 4 above)
Comment (irevoire) [reply meilisearch-lib/src/index_controller/dump_actor/mod.rs]: Done, I also splitted the code into multiple functions.

**Review by MarinPostma — APPROVED** [review]
lgtm

bors merge

**Pull request merged**